Source/WTF/wtf/NativePromise.h

```diff
@@ -1141,10 +1160,25 @@ class NativePromise final : public NativePromiseBase, public ConvertibleToNative
         // Only called by SettleFunction on the target's queue once all operations are complete and settled.
         // So we don't really need to hold the lock to access the value.
         Locker lock { m_lock };
-        ASSERT(!isNothing());
+        ASSERT(m_result.has_value());
         return *m_result;
     }
 
+    bool hasRunnable() const
+    {
+        Locker lock { m_lock };
+        return m_result.hasRunnable();
+    }
+
+    Function<Result(void)>& resultRunnable()
+    {
+        // Only called by SettleFunction on the target's queue once all operations are complete and settled.
+        // So we don't really need to hold the lock to access the value.
+        Locker lock { m_lock };
```

Comment:
Is this `m_lock` the same as the one that's `DropLockForScope`'d in the caller above?
If yes, to avoid the extra lockings (especially if they're not really necessary), could you instead call `hasRunnable()` and `resultRunnable()` before the `DropLockForScope`.
And maybe even combine them into one function that return a nullable pointer?

Reply:
The lock really does nothing, we return a reference here too.

Reply:
also, we do need a lock otherwise it will not compile due to the `WTF_GUARDED_BY_LOCK`

Reply:
I've tweaked things a bit so that if a copy/move is to occur, it happens under the lock.